core/src/replay_stage.rs

```diff
@@ -1232,6 +1237,14 @@ impl ReplayStage {
                         return false;
                     }
 
+
+                    // Should not purge slots for which we were the leader
+                    if let Some(leader_pubkey) = leader_schedule_cache.slot_leader_at(*duplicate_slot, None) {
+                        if leader_pubkey == *my_pubkey {
+                            panic!("We are attempting to dump a block that we produced. This indicates that we are producing duplicate blocks, or that there is a bug in our runtime/replay code causing us to compute different bank hashes than the rest of the cluster. We froze slot {} with hash {:?} while the cluster hash is {}", *duplicate_slot, frozen_hash, *correct_hash);
+                        }
+                    }
```

Comment:
Instead of nested `if` can you just do:
```rust
if Some(*my_pubkey) == leader_schedule_cache.slot_leader_at(*duplicate_slot, None) {
```

Can you please break the `panic` line to shorter lines using the `\` at line breaks?
similar to :
https://github.com/solana-labs/solana/blob/2b4a6a4dd/core/src/consensus.rs#L677

The panic description says "dump"; other places say "purge".
Would be good to use consistent terms throughout.

also the format arguments can be inlined:
https://rust-lang.github.io/rust-clippy/master/#uninlined_format_args

Reply:
>The panic description says "dump"; other places say "purge".
Would be good to use consistent terms throughout.

I agree we do this quite often in both replay and repair. I fixed up some occurrences here and created #30225 to track cleanup for the rest. Will discuss with carl and wen on what the proper convention should be.